src/main/java/org/folio/inventory/dataimport/consumers/MarcHoldingsHridSetKafkaHandler.java

```diff
@@ -0,0 +1,90 @@
+import static org.folio.rest.util.OkapiConnectionParams.OKAPI_TOKEN_HEADER;
+import static org.folio.rest.util.OkapiConnectionParams.OKAPI_URL_HEADER;
+
+/**
+ * This handler is a part of post-processing logic that is intended to handling Holdings creation.
+ * The handler is intended to process the event that is thrown on setting HRID to Holdings record
+ * Execution steps:
+ * 1. Create a new Holdings record by the way of mapping the incoming SRS Marc record using Mapping profile
+ * 2. Retrieve an existing Inventory Holdings record
+ * 3. Merge a new record (from step 1) with an existing record (from step 2)
+ * 4. Update an existing record (from step 2) with a result of merge (from step 3)
+ */
+public class MarcHoldingsHridSetKafkaHandler implements AsyncRecordHandler<String, String> {
+  private static final Logger LOGGER = LogManager.getLogger(MarcHoldingsHridSetKafkaHandler.class);
+  private static final String CORRELATION_ID_HEADER = "correlationId";
+  private static final String HOLDINGS_ID = "holdingsId";
+  private static final ObjectMapper OBJECT_MAPPER = ObjectMapperTool.getMapper();
+
+  private final HoldingUpdateDelegate holdingUpdateDelegate;
+  private final KafkaInternalCache kafkaInternalCache;
+
+  public MarcHoldingsHridSetKafkaHandler(HoldingUpdateDelegate holdingUpdateDelegate, KafkaInternalCache kafkaInternalCache) {
+    this.holdingUpdateDelegate = holdingUpdateDelegate;
+    this.kafkaInternalCache = kafkaInternalCache;
+  }
+
+  @Override
+  public Future<String> handle(KafkaConsumerRecord<String, String> record) {
+    try {
+      Promise<String> promise = Promise.promise();
+      Event event = OBJECT_MAPPER.readValue(record.value(), Event.class);
+      if (!kafkaInternalCache.containsByKey(event.getId())) {
+        kafkaInternalCache.putToCache(event.getId());
+        @SuppressWarnings("unchecked")
+        Map<String, String> headersMap = KafkaHeaderUtils.kafkaHeadersToMap(record.headers());
+        String correlationId = headersMap.get(CORRELATION_ID_HEADER);
+        LOGGER.info(format("Event payload has been received with event type: %s and correlationId: %s", event.getEventType(), correlationId));
+        DataImportEventPayload eventPayload = new JsonObject(ZIPArchiver.unzip(event.getEventPayload())).mapTo(DataImportEventPayload.class);
+
+        Context context = EventHandlingUtil.constructContext(headersMap.get(OKAPI_TENANT_HEADER), headersMap.get(OKAPI_TOKEN_HEADER), headersMap.get(OKAPI_URL_HEADER));
+        if (!(eventPayload.getContext().containsKey(HOLDINGS_ID) && eventPayload.getContext().containsKey(HOLDINGS.value()) && eventPayload.getContext().containsKey(MARC_HOLDINGS.value()))) {
+          throw new IllegalArgumentException(format("The event payload does not contain all the %s, %s, %s", HOLDINGS_ID, HOLDINGS.value(), MARC_HOLDINGS.value()));
+        }
+        String existingHoldingsId = eventPayload.getContext().get(HOLDINGS_ID);
+        MappingManager.map(eventPayload);
```

Comment:
What is the result of this operation?

Reply:
No result returned. The mapped record is put in a context, see the next line